homework-submission/myapp/src/Components/Data.js

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+import Todos from './Todos.json';
+import Form from './Form';
+
+
+class Data extends React.Component {
+  state = {
+    Todos
+  };
+
+  descriptionInput = React.createRef();
+  deadlineInput = React.createRef();
+
+  addTodo = event => {
+    event.preventDefault();
+
+    const descriptionInput = this.descriptionInput.current.value;
+    const deadlineInput = this.deadlineInput.current.value;
+    const newState = this.state;
+
+    const Todos = newState.Todos;
+    Todos.push({
+      id: Todos.length + 1,
+      description: descriptionInput,
+      deadline: deadlineInput,
+      done: false
+    });
+
+    this.setState(newState);
+
+    this.descriptionInput.current.value = '';
+    this.deadlineInput.current.value = '';
+  };
+
+  deleteTodo = id => {
+    const arr = this.state.Todos;
+    arr.splice(id, 1);
```

Comment:
same as a previous comment, this is mutating the state immediately, perhaps `.filter` Method ?

Remember the checking test:
```javascript
    console.log(this.state.Todos)
    arr.splice(id, 1);
    console.log(this.state.Todos)
```

Reply:
Hahaha i get what you mean. Probably `.splice()` method was easy for me to use. However thanks @Jawhar-B for the constructive feedback. i will try using `.filter()` method.